cmd/gardener-controller-manager/app/app.go

```diff
@@ -216,6 +217,41 @@ func run(ctx context.Context, log logr.Logger, cfg *controllermanagerconfigv1alp
 			}
 		}
 
+		managedSeeds := &seedmanagementv1alpha1.ManagedSeedList{}
+		if err := mgr.GetClient().List(ctx, managedSeeds); err != nil {
+			return fmt.Errorf("failed listing objects: %w", err)
+		}
+		if err := meta.EachListItem(managedSeeds, func(o runtime.Object) error {
```

Comment:
You already have the concrete object. Why not a simple `for` loop over `managedSeedList.Items`?

Reply:
https://github.com/gardener/gardener/pull/12029/commits/b8bdea331352b08715aa40601a380931ed7abe14